tests/Doctrine/Tests/DBAL/Connections/MasterSlaveConnectionTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Connections;
+
+use Doctrine\DBAL\Connections\MasterSlaveConnection;
+use Doctrine\DBAL\Driver;
+use Doctrine\Tests\DbalTestCase;
+
+class MasterSlaveConnectionTest extends DbalTestCase
+{
+    /**
+     * \Doctrine\DBAL\Connection::getParams has been marked @internal as of 2.11.0
+     * This test will be removed in 3.0
+     */
+    public function testConnectionDoesParamsRemainAvailable(): void
+    {
+        $constructionParams = [
+            'driver' => 'pdo_mysql',
+            'keepSlave' => true,
+            'master' => [
+                'host' => 'master.host',
+                'user' => 'root',
+                'password' => 'password',
+                'port' => '1234',
+            ],
+            'slaves' => [
+                [
+                    'host' => 'slave1.host',
+                    'user' => 'root',
+                    'password' => 'password',
+                    'port' => '1234',
+                ],
+            ],
+        ];
+
+        $connection = new MasterSlaveConnection($constructionParams, $this->createMock(Driver::class));
+
+        $connectionParams = $connection->getParams();
+        foreach ($constructionParams as $key => $value) {
+            self::assertSame($constructionParams[$key], $value);
```

Comment:
Did you check that the test fails if you revert your changes?

Reply:
Yes

Reply:
Really?

```
git restore --source=origin/2.11.x lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php
vendor/bin/phpunit tests/Doctrine/Tests/DBAL/Connections/MasterSlaveConnectionTest.php                       
PHPUnit 9.3.2 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 00:00.030, Memory: 6.00 MB

OK (1 test, 4 assertions)
```

Any idea why I'm commenting on this specific line? :smile:

Reply:
The test should use `$connectionParams[$key]` on that line. Otherwise, it is comparing an array to itself.

Reply:
Should have been in a spoiler tag :stuck_out_tongue:

Reply:
oops, i refactored before my last push and broke, fixed now

Reply:
It still looks wrong. Did you forget to push or something?

Reply:
was just waiting for composer / phpunit / phpcs locally etc before pushing